src/table/filter/standard_bloom/mod.rs

```diff
@@ -59,11 +59,27 @@ impl<'a> StandardBloomFilterReader<'a> {
         let hash_type = reader.read_u8()?;
         assert_eq!(0, hash_type, "Invalid bloom hash type");
 
+        #[expect(
+            clippy::cast_possible_truncation,
+            reason = "truncation is not expected to happen"
```

Comment:
A lot of these reasonings are basically stating the obvious, but, for example, in this case, really the reasoning is: `there cannot be a filter that is 4 GiB in size` (on 32 bit targets) - filters of a single table never even approach multiple megabytes in size typically.